source/common/http/match_delegate/config.cc

```diff
@@ -105,8 +105,9 @@ void DelegatingStreamFilter::FilterMatchState::evaluateMatchTree(
     return;
   }
 
-  // If no match tree is set, interpret as a skip.
-  if (!has_match_tree_) {
+  // If no match tree is set or it is disabled by runtime guard, interpret as a skip.
+  if (!has_match_tree_ ||
+      !Runtime::runtimeFeatureEnabled("envoy.reloadable_features.enable_extension_with_matcher")) {
```

Comment:
These feature flags aren't designed as permanent config knobs, and I feel this capability should exist generally. Also, you might want to do the runtime override on a per match filter basis.

I'd suggest having an API in which we add `runtime_key` (search for other users in Envoy API) to allow the runtime key to be specific. That way you can have individual filter control.

As discussed the other day, adding a disabling filter might be more general, since this is a useful feature in general for filters, the idea that we might want to with runtime turn on/off a specific filter in the filter chain. Such filters could be wrapper with a RuntimeDisabledFilter filter. This is more an optional nicety, I'm good with just adding `runtime_key` to the config.

CC @envoyproxy/api-shepherds

Reply:
Recent work (#25927) added the [disabled](https://github.com/envoyproxy/envoy/blob/3007db0651fe4732751efe6a4239b1a02bb6e05f/api/envoy/config/route/v3/route_components.proto#L2415) knob for an http filter. Is it sufficient, or is a runtime key also required for this use-case?

Reply:
Re: @htuch , I thought we just need a temporary runtime flag. I agree that runtime key will be a permanent knob and enables per-filter granularity and other flexibilities.  Will update.

Re @adisuissa: We need runtime control here. That [disabled](https://github.com/envoyproxy/envoy/blob/3007db0651fe4732751efe6a4239b1a02bb6e05f/api/envoy/config/route/v3/route_components.proto#L2415) knob still needs the config updates, IIUC.

Reply:
Yes, it does require a config update.
If you need a [RuntimeFeatureFlag](https://github.com/envoyproxy/envoy/blob/3007db0651fe4732751efe6a4239b1a02bb6e05f/api/envoy/config/core/v3/base.proto#L291) field, consider adding it next to the `disabled` field.